### DISCUSSION TITLE
Organic and manual step counting

### DISCUSSION THREAD

**Discussion by Narvaez0993** [matinzd/react-native-health-connect#170] [Q&A] [answered]
Hello, I am integrating Health Connect into my application for step counting, but I need to find a way to exclude manually recorded steps. When retrieving the steps, is there any way to differentiate between organic steps and those manually entered by apps like Google Fit?

I thought about the metadata returned with the step information, but when entering manual data from Google Fit and recording organic steps, I can't find any flag to identify which steps were recorded manually and which were generated organically by the user.

**Every time I query steps in Health Connect like this:**

`const steps = await readRecords('Steps', { timeRangeFilter });`

I receive the following in the metadata:

{
  "clientRecordId": null,
  "clientRecordVersion": 0,
  "dataOrigin": "com.google.android.apps.fitness",
  "device": null,
  "id": "63bd49f0-6bfd-41ae-99f7-e13e44ec21f3",
  "lastModifiedTime": "2024-10-11T18:40:11.181Z",
  "recordingMethod": 0
}

Is there any way to achieve this?

**Comment by matinzd** [comment, marked as answer]
Duplicate of https://github.com/matinzd/react-native-health-connect/issues/51

**Reply by Narvaez0993** [reply]
Thank you very much, @matinzd
